include/envoy/server/factory_context.h

```diff
@@ -152,6 +152,13 @@ class ServerFactoryContext : public virtual CommonFactoryContext {
    * @return StatsConfig& the servers stats configuration.
    */
   virtual StatsConfig& statsConfig() PURE;
+
+  /**
+   * @return AccessLogManager for use by the entire server.
+   */
+  virtual AccessLog::AccessLogManager& accessLogManager() PURE;
+
+  virtual ProtobufMessage::ValidationVisitor& messageValidationVisitor() PURE;
```

Comment:
Per other comment do we really need this change?

Reply:
Can you move these 2 methods into CommonFactoryContext (and remove from FactoryContext) and then have the access logger take CommonFactoryContext for now? I think this should work based on my read of the code and will give us a bit more type checking in the future if we do anything that makes the access loggers more independent of the server context?

Reply:
Mostly fixed in b8c57d538b2084b98a42d4ad5a92645c7f27b3d0